agent/connectionuptime/service.go

```diff
@@ -141,6 +141,9 @@ func (c *Service) RunCleanupGoroutine(ctx context.Context) {
 //
 //	time_between(f1, now) - total time between first event (f1) and current moment
 func (c *Service) GetConnectedUpTimeUntil(toTime time.Time) float32 {
+	c.mx.Lock()
+	defer c.mx.Unlock()
+
```

Comment:
I noticed that we don't change any field here, so we can change mutex type to RWMutex and use RLock here.